pkg/apis/shared/namevalue.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shared
+
+// +kubebuilder:object:generate=true
+
+import (
+	"strconv"
+
+	apiextensionsV1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// NameValuePair is a name/value pair.
+type NameValuePair struct {
+	// Name of the property.
+	Name string `json:"name"`
+	// Value of the property, in plaintext.
+	//+optional
+	Value DynamicValue `json:"value,omitempty"`
+	// SecretKeyRef is the reference of a value in a secret store component.
+	//+optional
+	SecretKeyRef SecretKeyRef `json:"secretKeyRef,omitempty"`
+	// EnvRef is the name of an environmental variable to read the value from.
+	//+optional
+	EnvRef string `json:"envRef,omitempty"`
+}
+
+// HasValue returns true if the NameValuePair has a non-empty value.
+func (nvp NameValuePair) HasValue() bool {
+	return len(nvp.Value.JSON.Raw) > 0
+}
```

Comment:
This should be checking a nil pointer, rather than the presence of string. An empty string should be considered a value.

Reply:
`len()` does a nil-check built-in. With our CRDs (also Configuration etc), we consider empty strings the same as non-present values and we don't have a need to distinguish between the two.